packages/mds-compliance-service/repository/index.ts

```diff
@@ -269,9 +279,143 @@ class ComplianceReadWriteRepository extends ReadWriteRepository {
     }
   }
 
+  public getComplianceViolation = async (
+    options: GetComplianceViolationOptions
+  ): Promise<ComplianceViolationDomainModel> => {
+    // TODO look for cleaner solution
+    const isViolationIdOption = (option: unknown): option is { violation_id: UUID } =>
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (option as any).violation_id
+
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      if (isViolationIdOption(options)) {
+        const { violation_id } = options
+        const entity = await connection.getRepository(ComplianceViolationEntity).findOne({
+          where: {
+            violation_id
+          }
+        })
+        if (!entity) {
+          throw new NotFoundError(`ComplianceViolation ${violation_id} not found`)
+        }
+        return ComplianceViolationEntityToDomain.map(entity)
+      }
+
+      const { event_timestamp, device_id, trip_id } = options
+      if (!isDefined(event_timestamp) || !isDefined(device_id)) {
+        throw RepositoryError('event_timestamp and device_id must be given if violation_id is not')
+      }
+      let query = connection
+        .getRepository(ComplianceViolationEntity)
+        .createQueryBuilder()
+        .where('event_timestamp = :event_timestamp', { event_timestamp })
+        .andWhere('device_id = :device_id', { device_id })
+      if (isDefined(trip_id)) {
+        query = query.andWhere('trip_id = :trip_id', { trip_id })
```

Comment:
Do you need to redefine `query` here? Seeing mixed usage throughout this PR of redefining `query`, versus just calling `query.someMethod()`.